CMakeLists.txt

```diff
@@ -44,13 +44,21 @@ set(SOVERSION 1)
 set(CMAKE_BUILD_TYPE  Debug CACHE STRING "Choose the type of build" FORCE)
 set(BUILD_SHARED_LIBS OFF   CACHE BOOL   "Whether or not to build shared libraries instead of static")
 
+add_definitions(-DLEGION_CMAKE)
+
 #https://github.com/StanfordLegion/legion/issues/168#issuecomment-244582958
 include(CheckCXXCompilerFlag)
 check_cxx_compiler_flag("-fno-strict-aliasing" COMPILER_SUPPORTS_NO_STRICT_ALIASING)
 if(COMPILER_SUPPORTS_NO_STRICT_ALIASING)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing")
 endif()
 
+if (${CMAKE_BUILD_TYPE} EQUAL "Debug")
+  add_definitions(-DDEBUG_REALM -DDEBUG_LEGION)
```

Comment:
@junghans:  legion_defines.h gets generated after:
add_definitions(-DDEBUG_REALM -DDEBUG_LEGION)

I guess instead of "add_definitions" I should do "set (DEBUG_REALM ON)" ?